extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/util/SqlStatementResourceHelper.java

```diff
@@ -221,16 +216,24 @@ private static Optional<List<PageInformation>> populatePagesForDurableStorageDes
       throw DruidException.defensive("Expected worker count to be set for stage[%d]", finalStage);
     }
 
-
     List<PageInformation> pages = new ArrayList<>();
+
+
+    Set<Integer> emptyWorkerCounters = new HashSet<>();
+
     for (int partitionNumber = 0; partitionNumber < totalPartitions; partitionNumber++) {
       for (int workerNumber = 0; workerNumber < totalWorkerCount; workerNumber++) {
         CounterSnapshots workerCounter = workerCounters.get(workerNumber);
+        if ((workerCounter == null || workerCounter.isEmpty()) && !emptyWorkerCounters.contains(workerNumber)) {
+          pages.add(new PageInformation(pages.size(), 0L, 0L, workerNumber, null));
+          emptyWorkerCounters.add(workerNumber);
```

Comment:
I donot understand why we require a hashSet here. 
Also, DurableStorageDestination is only for selects and not inserts. Not sure if changes are required here.

Reply:
Got it. Reverted the changes for `DurableStorageDestination` type as we want to report 0 value counters only for ingest queries